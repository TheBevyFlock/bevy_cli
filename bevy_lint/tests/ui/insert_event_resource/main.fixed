#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_event_resource)]

use bevy::prelude::*;

#[derive(Event)]
struct Foo;

fn main() {
    App::new().add_event::<Foo>();
    //~^ ERROR: called `App::init_resource::<Events<T>>()` instead of `App::add_event::<T>()`

    App::new().add_event::<Foo>();
    //~^ ERROR: called `App::insert_resource(Events<T>)` instead of `App::add_event::<T>()`

    // Make sure the correct type is detected, even when not explicitly passed to
    // `insert_resource()`.
    let implied_event: Events<Foo> = Default::default();
    App::new().add_event::<Foo>();
    //~^ ERROR: called `App::insert_resource(Events<T>)` instead of `App::add_event::<T>()`

    // Ensure the lint can be muted by annotating the expression.
    #[allow(bevy::insert_event_resource)]
    App::new().init_resource::<Events<Foo>>();
}
