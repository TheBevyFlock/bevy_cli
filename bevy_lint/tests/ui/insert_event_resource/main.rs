#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_event_resource)]

use bevy::prelude::*;

#[derive(Event)]
struct Foo;

fn main() {
    App::new().init_resource::<Events<Foo>>();
    //~^ ERROR: called `App::init_resource::<Events<T>>()` instead of `App::add_event::<T>()`

    App::new().insert_resource::<Events<Foo>>(Default::default());
    //~^ ERROR: called `App::insert_resource(Events<T>)` instead of `App::add_event::<T>()`

    // Make sure the correct type is detected, even when not explicitly passed to
    // `insert_resource()`.
    let implied_event: Events<Foo> = Default::default();
    App::new().insert_resource(implied_event);
    //~^ ERROR: called `App::insert_resource(Events<T>)` instead of `App::add_event::<T>()`

    // Ensure the lint can be muted by annotating the expression.
    #[allow(bevy::insert_event_resource)]
    App::new().init_resource::<Events<Foo>>();
}
