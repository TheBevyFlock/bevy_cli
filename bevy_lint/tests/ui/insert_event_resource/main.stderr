error: called `App::init_resource::<Events<Foo>>()` instead of `App::add_event::<Foo>()`
  --> tests/ui/insert_event_resource/main.rs:12:16
   |
12 |     App::new().init_resource::<Events<Foo>>();
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/insert_event_resource/main.rs:3:9
   |
3  | #![deny(bevy::insert_event_resource)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: inserting an `Events` resource does not fully setup that event
   |
12 -     App::new().init_resource::<Events<Foo>>();
12 +     App::new().add_event::<Foo>();
   |

error: called `App::init_resource::<Events<Foo>>(&mut app)` instead of `App::add_event::<Foo>(&mut app)`
  --> tests/ui/insert_event_resource/main.rs:15:5
   |
15 |     App::init_resource::<Events<Foo>>(&mut app);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Events` resource does not fully setup that event
   |
15 -     App::init_resource::<Events<Foo>>(&mut app);
15 +     App::add_event::<Foo>(&mut app);
   |

error: called `App::insert_resource::<Events<Foo>>(Default::default())` instead of `App::add_event::<Foo>()`
  --> tests/ui/insert_event_resource/main.rs:19:16
   |
19 |     App::new().insert_resource::<Events<Foo>>(Default::default());
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Events` resource does not fully setup that event
   |
19 -     App::new().insert_resource::<Events<Foo>>(Default::default());
19 +     App::new().add_event::<Foo>();
   |

error: called `App::insert_resource::<Events<Foo>>(&mut app, Default::default())` instead of `App::add_event::<Foo>(&mut app)`
  --> tests/ui/insert_event_resource/main.rs:23:5
   |
23 |     App::insert_resource::<Events<Foo>>(&mut app, Default::default());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Events` resource does not fully setup that event
   |
23 -     App::insert_resource::<Events<Foo>>(&mut app, Default::default());
23 +     App::add_event::<Foo>(&mut app);
   |

error: called `App::insert_resource(implied_event)` instead of `App::add_event::<Foo>()`
  --> tests/ui/insert_event_resource/main.rs:30:16
   |
30 |     App::new().insert_resource(implied_event);
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Events` resource does not fully setup that event
   |
30 -     App::new().insert_resource(implied_event);
30 +     App::new().add_event::<Foo>();
   |

error: aborting due to 5 previous errors

